src/test/java/spoon/test/invocations/InvocationTest.java

```diff
@@ -58,4 +62,32 @@ public void testTargetNullForStaticMethod() throws Exception {
 			fail();
 		}
 	}
+
+	@Test
+	public void testInvocationWithArgument() throws Exception {
+		Launcher launcher = new Launcher();
+		launcher.addInputResource("./src/test/java/spoon/test/invocations/testclasses/Foo.java");
+		launcher.buildModel();
+
+		CtClass<?> foo = launcher.getFactory().Class().get(Foo.class);
+		CtConstructor<?> constructor = foo.getConstructors().iterator().next();
+
+		CtInvocation<?> invocBar = constructor.getBody().getLastStatement();
+
+		assertEquals(3, invocBar.getArguments().size());
+		CtExpression secondArg = invocBar.getArguments().get(1);
+
+		assertTrue(secondArg instanceof CtLiteral);
+
+		CtLiteral arg = (CtLiteral)secondArg;
+		assertEquals(42, arg.getValue());
+
+		secondArg = launcher.getFactory().Core().createLiteral().setValue(12);
+
+		invocBar.addArgument(1, secondArg);
```

Comment:
I do not agree with you.
Spoon is not designed to check to validity of the transformation, it is not realistic to have always consistant model.

we can create a scanner that check the consistant of the model but it is not the responsibility of the setters.